File: api/src/main/scala/uk/ac/surrey/xw/api/swing/LogoObjectListCellRenderer.scala
package uk.ac.surrey.xw.api.swing

import org.nlogo.api.Dump

import javax.swing.DefaultListCellRenderer
import javax.swing.JList

class LogoObjectListCellRenderer extends DefaultListCellRenderer {
  override def getListCellRendererComponent(
    list: JList, value: AnyRef, index: Int,
    isSelected: Boolean, cellHasFocus: Boolean) = {
    super.getListCellRendererComponent(list, value, index,
      isSelected, cellHasFocus)
    setText(Option(value).map(Dump.logoObject).getOrElse(""))
    this
  }
}

File: api/src/main/scala/uk/ac/surrey/xw/api/swing/LogoObjectListCellRenderer.java
package uk.ac.surrey.xw.api.swing;

import java.awt.Component;

import javax.swing.DefaultListCellRenderer;
import javax.swing.JList;

import org.nlogo.api.Dump;

@SuppressWarnings("serial")
public class LogoObjectListCellRenderer extends DefaultListCellRenderer {

  @Override
  public Component getListCellRendererComponent(JList<? extends Object> list, Object value,
      int index, boolean isSelected, boolean cellHasFocus) {
    super.getListCellRendererComponent(list, value, index, isSelected, cellHasFocus);
    setText(value == null ? "" : Dump.logoObject(value));
    return this;
  }

}
